src/rezplugins/shell/_utils/powershell_base.py

```diff
@@ -105,8 +106,11 @@ def get_syspaths(cls):
         return cls.syspaths
 
     def _bind_interactive_rez(self):
+        curr_prompt = os.getenv("REZ_CURRENT_PROMPT", "")
```

Comment:
That's a good point. I was mostly looking at the cmd plugin. In that the REZ_STORED_PROMPT_CMD is **not** the current prompt but the one of the parent shell. See https://github.com/nerdvegas/rez/blob/8a1af9724118db54cc81f41557272374d0099558/src/rezplugins/shell/cmd.py#L146

From a quick look it seems to be the same in `sh` and `csh`. But it looks to me like none of this is ever used?

But as STORED_PROMPT refers to the prompt of the parent shell i went with CURRENT instead.

Not sure i got the setup totally right, nor do i really understand whiy this is the case

Reply:
No it looks like you should use REZ_STORED_PROMPT_<type> also. In cwd, this is just getting set to current prompt if it hasn't been already set in a parent rez shell, that's all.

There's a problem with using REZ_CURRENT_PROMPT too - it's not shell-specific, which could cause weird cross-shell prompts if another shell type also decided to use REZ_CURRENT_PROMPT (ie, nothing is stopping you opening a bash rez shell within a tcsh rez shell within a pwsh rez shell. It's weird and an edge case that may never happen, but still).

Reply:
I don't see a problem with changing the name. However i think this might be broken or non-canonical in cmd (and subsequently in PowerShells) or i just don't get it yet. If i spawn 3 nested shells in cmd this is what i get:

- Prompt is prefixed with `>>>` as expexted
- REZ_ENV_PROMPT is set to $G$G$G ($G represenents `>` when used in the `PROMPT` environment variable (wich is setting the prompt when set)
- REZ_STORED_PROMPT_CMD is set to $P$G ($P represents current drive and path when used int the `PROMPT` environment variable
- PROMPT (which sets the actual prompt in CMD) is set to `$G$G$G $P$G` as expected (which reflects the 3 greater signs prefixing the original default prompt setting)

So in CMD `REZ_STORED_PROMPT_CMD` does not reflect the current prompt as set by Rez, but the prompt of the original shell that rez was first called from. It is set only when not set in the shell already which is only true in the first rez shell. All sub-shells never touch it again as it is already set. 

After writing the above, I did just check and it seems to be the same for bash. I think i still don't really understand, but i can mimick that behaviour and just add a disable_set_prompt as proposed by you below. It feels a bit counterintuitive to me to have a global setting called `set_prompt` and a shell specific one `disable_set_prompt` that only disables parts of the former. But i don't have a strong opinion either.

Reply:
Ok I remember now. Yes, REZ_STORED_PROMPT_* is just storing the original non-rez prompt. It just does this so it can construct the current prompt properly, which is usually (REZ_ENV_PROMPT + REZ_STORED_PROMPT).

Wrt `set_prompt`, I completely forgot that was there! So... what is `set_prompt_as_env` for then? I think I'm missing something.

Reply:
I probably should have been more explicit in the original description. Currently the CMD setup just sets %PROMPT% which cmd uses as the prompt.

For Powershell however, the prompt is a function. So you define a function called prompt{} that returns the prompt.
The shell plugin just sets this. So the current implementation does no show nesting at all. Only if you are in a rez shell or not.

In addition there are many 3rd party prompt tools available (oh-my-posh being a popular one, that you can see in the screenshots above). They basically implement fancy prompt{} functions to display all sorts of info in fancy ways. When you use any of them, resolving a rez shell just wipes your prompt and replaces it with the standard rez prompt.

The idea behind the implementation behind this PR is basically:

- global `set_prompt` defines if you want the prompt to be set
- `set_prompt_as_env` controls for powershell based shells if the prompt{} function is actually replaced (which is current behaviour) or (when set to true) rez only sets the prompt as an environment variable without actually replacing prompt{}

In the screenshots above this is set to true, and then i am using the environment variable in my oh-my-posh template to 
inject the rez shell into the custom prompt provided by oh-my-posh